code/game/machinery/doors/airlock.dm

```diff
@@ -348,14 +354,20 @@ GLOBAL_LIST_EMPTY(airlock_overlays)
 	var/image/sparks_overlay
```

Comment:
This proc is such a mess. Outside of this PRs scope though

Reply:
Respectfully disagree. This is the engine behind all airlock sprite changes, I replicated the procedure used with overlays for underlays exactly. I don't know of a better way of accomplishing consistent and cost effective alterations of airlocks sprites so that the correct underlays are used and updated in the exact same timing as the airlock sprites.

If this proc is out of scope then I will have to trash all airlock emissives, I really dont want to do that.

Reply:
It's the proc being a mess and fixing it that's out of scope.